R/stan_betareg.fit.R

```diff
@@ -20,7 +20,7 @@
 
 stan_betareg.fit <- function (x, y, z = NULL, weights = rep(1, NROW(x)), offset = rep(0, NROW(x)),
                               link = c("logit", "probit", "cloglog", "cauchit", "log", "loglog"), 
-                              link.phi = c("log", "identity", "sqrt"), ...,
+                              link.phi = "log", ...,
```

Comment:
Your comments aren't confusing. It's just that betareg specifies the model in a peculiar way and I'm using differing link.phi lengths and "|" to capture that info and pass it on to `stan_betareg.fit`. For instance, I need to distinguish between,
```
betareg(y ~ x , data = dat, link = "log")
```
and
```
betareg(y ~ x , data = dat, link = "log", link.phi = "log")
betareg(y ~ x | 1, data = dat, link = "log")
betareg(y ~ x | 1, data = dat, link = "log", link.phi = "log")
```
So going back to what you said in a previous comment:

> ... it's only unnecessary but not wrong for a user to specify link.phi="log" or whatever single value even if there's no Z.

If there is no Z specified by the user then the type of model fit if link.phi is not specified by the user is different than if the user does specify link.phi. So,
```
betareg(y ~ x , data = dat, link = "log")
betareg(y ~ x , data = dat, link = "log", link.phi = "log")
```
are different models. 

---
But, given your comments about `match.arg`, in `stan_betareg` I can do 
```
 if (length(grep("\\|", all.names(formula))) == 1 && length(link.phi) > 1) {
    link.phi <- match.arg(link.phi)
  }
```
and in `stan_betareg.fit` I can do
```
  if (length(link.phi) > 1 || is.null(z)) {
    Z_true <- 0
    z <- model.matrix(y ~ 1)
    link.phi <- match.arg(link.phi)
  }
  else if (length(link.phi) == 1) {
    Z_true <- 1
  }
```

---
And I can pass the `match.arg` statements for `link` and `link.phi` in stan_betareg.fit before the model fitting happens (regardless though match.arg is called towards the beginning of stan_betareg.fit). 

Sorry that this explanation is so long-winded.